sheetstorm/tests/test_unit_generate_test_data.py

```diff
@@ -0,0 +1,178 @@
+        self.assertTrue(Command._get_superuser_request({self.SUPERUSER: True}))
+
+    def test_function_should_return_false_if_not_is_superuser_request(self):
+        self.assertFalse(Command._get_superuser_request({self.SUPERUSER: False}))
+
+    @parameterized.expand(
+        [CustomUser.UserType.ADMIN.name, CustomUser.UserType.EMPLOYEE.name, CustomUser.UserType.MANAGER.name, SUPERUSER]
+    )
+    def test_only_admins_and_superuser_should_have_staff_property(self, user_type):
+        if user_type in (CustomUser.UserType.ADMIN.name, self.SUPERUSER):
+            self.assertTrue(Command._set_user_factory_parameters(user_type).get("is_staff"))
+        else:
+            self.assertFalse(Command._set_user_factory_parameters(user_type).get("is_staff"))
+
+    @parameterized.expand(
+        [CustomUser.UserType.ADMIN.name, CustomUser.UserType.EMPLOYEE.name, CustomUser.UserType.MANAGER.name, SUPERUSER]
+    )
+    def test_only_superuser_should_have_superuser_property(self, user_type):
+        if user_type == self.SUPERUSER:
+            self.assertTrue(Command._set_user_factory_parameters(user_type).get("is_superuser"))
+        else:
+            self.assertFalse(Command._set_user_factory_parameters(user_type).get("is_superuser"))
+
+    @parameterized.expand(
+        [CustomUser.UserType.ADMIN.name, CustomUser.UserType.EMPLOYEE.name, CustomUser.UserType.MANAGER.name, SUPERUSER]
+    )
+    def test_only_admins_and_superuser_should_have_user_type_admin(self, user_type):
+        if user_type in (CustomUser.UserType.ADMIN.name, self.SUPERUSER):
+            self.assertEqual(
+                Command._set_user_factory_parameters(user_type).get("user_type"), CustomUser.UserType.ADMIN.name
+            )
+        else:
+            self.assertNotEqual(
+                Command._set_user_factory_parameters(user_type).get("user_type"), CustomUser.UserType.ADMIN.name
+            )
+
+    @parameterized.expand(
+        [CustomUser.UserType.ADMIN.name, CustomUser.UserType.EMPLOYEE.name, CustomUser.UserType.MANAGER.name, SUPERUSER]
+    )
+    def test_result_of_create_user_function_should_be_specified_number_of_users_in_database(self, user_type):
+        Command(CustomCommandOptions).create_user(user_type, CustomCommandOptions.number_of_users_to_create)
+        if user_type == CustomUser.UserType.ADMIN.name:
+            self.assertEqual(
+                CustomUser.objects.filter(user_type=user_type, is_superuser=False).count(),
+                CustomCommandOptions.number_of_users_to_create,
+            )
+        elif user_type == self.SUPERUSER:
+            self.assertEqual(
+                CustomUser.objects.filter(is_superuser=True).count(), CustomCommandOptions.number_of_users_to_create
+            )
+        else:
+            self.assertEqual(
+                CustomUser.objects.filter(user_type=user_type).count(), CustomCommandOptions.number_of_users_to_create
+            )
+
+
+class CreateProjectMethodsTests(TestCase):
```

Comment:
Any counterparts of missing tests that I described under the user's tests that aren't applied here: they also apply here 😛

Reply:
I hope I done it well :)

Reply:
I hope I applied all your suggestions :)